broker/service/api/v10.py

```diff
@@ -153,3 +163,378 @@ def submission_log(submission_id):
     stdout.close()
 
     return logs
+
+
+""" Gets the visualizer url of a specific job.
+
+Raises:
+    ex.BadRequestException -- Trying to search info about a job that
+    has never being submitted in this Asperathos instance.
+
+Returns:
+    dict -- Returns a dict with 'visualizer_url' as key and the url
+            that gives access to the visualizer platform as value.
+"""
+
+def submission_visualizer(submission_id):
+
+    if submission_id not in submissions.keys():
+        API_LOG.log("Wrong request")
+        raise ex.BadRequestException()
+
+    visualizer_url = ""
+
+    # Check if the visualizer is active in this Asperathos instance
+    # If true, call visualizer API to return the visualizer URL
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+    address = api.visualizer_url.split('/')[-1]
+    ip = address.split(':')[0]
+    port = int(address.split(':')[1])
+    
+    result = sock.connect_ex((ip, port))
+    if result == 0:
+        API_LOG.log("Visualizing Running on port %s" % port)
+        visualizer_url = visualizer.get_visualizer_url(api.visualizer_url, submission_id)
+    else:
+        API_LOG.log("There is no process running in the Visualizer address")
+
+    return {"visualizer_url": visualizer_url}
+
+""" Add a new cluster that can be choose to be the active
+   cluster in the Asperathos section in execution time.
+
+Raises:
+    ex.BadRequestException -- Missing cluster and authentication fields in request
+    ex.UnauthorizedException -- Wrong authentication variables informed
+
+Returns:
+    dict -- Returns a dict with the cluster_name, 
+    the status of the addition (success or failed) and a
+    reason in case of 'failed' status
+"""
+
+def add_cluster(data):
+    if ('cluster_name' not in data or 'cluster_config' not in data):
+        API_LOG.log("Missing cluster fields in request")
+        raise ex.BadRequestException("Missing cluster fields in request")
+
+    if ('enable_auth' not in data):
+        API_LOG.log("Missing parameters in request")
+        raise ex.BadRequestException()
+
+    enable_auth = data['enable_auth'] 
+
+    if enable_auth:
+        if 'username' not in data or 'password' not in data:
+            API_LOG.log("Missing parameters in request")
+            raise ex.BadRequestException()
+
+        username = data['username']
+        password = data['password']
+
+        authorization = authorizer.get_authorization(api.authorization_url,
+                                                 username, password)
+                                            
+        if not authorization['success']:
+            API_LOG.log("Unauthorized request")
+            raise ex.UnauthorizedException()
+
+    try:
+        controller.add_cluster(api.controller_url, data)
```

Comment:
this a huge burden to the overall execution of the framework. I think we should share the datastore where cluster definitions are written to between the components instead of triggering this sort of operation in all of them. also, by doing this we're adding a large point of failure and providing the user no useful information whatsoever (remember they are not required to use all components at the same time).

Reply:
Thanks for your feedback about this issue. You're right, this issue was a huge burden for the execution of the framework. This problem has been solved in the commit [540b521](https://github.com/ufcg-lsd/asperathos-manager/commit/540b521953feea4e73758f6d8173a1c58e9e65be)